release-24.1: server/license: Flow cluster init timestamp through tenant connector

spilchen opened the pull request:
Backport 1/1 commits from #131750.

/cc @cockroachdb/release

---

Currently, secondary tenants cannot access the cluster initialization grace period timestamp in the KV store because they lack permission to access the system keyspace. Instead of reading it directly from the KV, we now pass the timestamp through the tenant connector, making it accessible to secondary tenants. The timestamp is only written once during the system tenant’s initial startup, so it remains a constant value from the perspective of secondary tenants. The value is sent from the system tenant to secondary tenants using the existing TenantSettingsRequest for METADATA_EVENT.

This change will be backported to versions 24.2, 24.1, 23.2, and 23.1.

Epic: CRDB-39988
Closes CRDB-42309
Release note: none
Release justification: This work is part of the CockroachDB core deprecation.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132223)
<!-- Reviewable:end -->

spilchen committed:
server/license: Flow cluster init timestamp through tenant connector

Currently, secondary tenants cannot access the cluster initialization
grace period timestamp in the KV store because they lack permission to
access the system keyspace. Instead of reading it directly from the KV,
we now pass the timestamp through the tenant connector, making it
accessible to secondary tenants. The timestamp is only written once
during the system tenant’s initial startup, so it remains a constant
value from the perspective of secondary tenants. The value is sent from
the system tenant to secondary tenants using the existing
TenantSettingsRequest for METADATA_EVENT.

This change will be backported to versions 24.2, 24.1, 23.2, and 23.1.

Epic: CRDB-39988
Closes CRDB-42309
Release note: none
